src/internal/channel.js

```diff
@@ -113,7 +113,7 @@ export function channel(buffer) {
 }
 
 export function eventChannel(subscribe, buffer = buffers.none(), matcher) {
-  if(arguments.length > 1)
+  if(typeof matcher !== 'undefined')
```

Comment:
I understand, but in the above function there is no default value for the `matcher` argument. If I do the check only if `matcher` is not undefined, then it implies that this call form is allowed `eventChannel(subscribe, buffer, undefined)`. I dont think there is a case where a developer will want to pass undefined intentionally (except for setting the arg to the default value, which  is not the case above since match has no default value). If we got an undefined value it's likely  because of some typo error.